### DISCUSSION TITLE
Bad scan quality only on NAPS2 with Linux

### DISCUSSION THREAD

**Discussion by cloudmaker97** [cyanfish/naps2#155] [Support]
I am using Debian 12 (Bookworm) and trying to use a HP DeskJet 4100 _(Printer/ Scanner)_. The printing works as expected and like on other systems but the scanning have some weird problems:

- The ADF-Scan is working over the hplip SANE driver but it is scanned with really bad quality on NAPS.  
- The scan result is about approx. 200 DPI with 1200 DPI setting on DIN A4 document format
- When i using the built in `Document-Scanner` shipped with Debian the scan looks really great, but it isn't a real long-term solution for me because it doesn't come with such handy features like NAPS and the workflow with it is really slow.

The drivers `hplip` and `hplip` are installed from the package repositories and configured via the `hp-toolbox` GUI. If needed i can provide a example scan result. Just ask me for that. 

### System Details

```
OS: Debian GNU/Linux 12 (bookworm) x86_64 
Host: B550M AORUS ELITE 
Kernel: 6.1.0-9-amd64 
Shell: bash 5.2.15 
Resolution: 1920x1080, 1920x1080, 1920x1080 
DE: Cinnamon 5.6.8 Terminal: gnome-terminal 
CPU: AMD Ryzen 5 5600G with Radeon Graphics (12) @ 3.900GHz 
GPU: NVIDIA GeForce GTX 1660 Ti 
Memory: 7749MiB / 48018MiB 
```

### Registered CUPS-printer

- `hp:/net/DeskJet_4100_series?ip=192.168.178.35`
- `ipp://192.168.178.35:631/ipp`

### SANE-Devices on NAPS2

![image](https://github.com/cyanfish/naps2/assets/4189795/dd68e978-36c7-4989-90ca-097647461768)

### Network Details

- Computer connecting to the printer is on IP `10.0.100.10/8`
- Printer connected to a router is on `192.168.178.35/24`
- Between the computer and the printer is a Firewall on `10.0.0.1/8`
- The routing from the firewall and the printer is configured and all ports are open and accessible

-------

I hope that someone can help me to solve this problem. Thanks in advance! 😎

**Comment by cyanfish** [comment]
NAPS2 with SANE currently has an issue where if you select a DPI higher than your scanner supports it will ignore the setting. Try choosing 600dpi instead of 1200.

**Reply by cloudmaker97** [reply]
Oh yes, baby! That way worked! 🥳 
I had to turn it down to 300 DPI, didn't noticed a good result on 600 DPI.

----

Just a little side-note (don't know if it's worth a bug request or is related to the SANE issue), on the bad-quality scans i've noticed, that when i want to view a single page it shows me the page in a minium scale and dont want to fit correctly. Only on the good-quality scans it shows me the document in the best scale. But ... at least for me it works now and i am very happy.

![image](https://github.com/cyanfish/naps2/assets/4189795/6ef00e00-333f-41b4-8947-aec0373cfa25)
